improve: UI, Styles, MiniWindow General, see Roadmap

libergod opened the pull request:
# Description

- Introduced a new miniwindow header image for UI.
- Created a new style file for the miniwindow battle interface, defining layout, buttons, and scrollable content.
- Configured button states for close, minimize, and toggle filter functionalities.
- Established visual elements such as borders, backgrounds, and dimensions for a cohesive design.
- Multi-Instance battle list feature.
- Filters and Orderings for each instance.
- Lock buttons specific to each instance.
- Keep open/re-launch when relog/exit game.
- On close, it destroys the instance, cleaning up its settings.
- A bunch of safety checks and locks to ensure they are working properly, and update all windows accordingly.

## Roadmap
- [X] Battle List
   - [X] Filter Button, Filters like Tibia RL, with Monk Button ready to be used after version 15.
   - [X] Sorts Asc/Desc
   - [X] Edit Window Name
   - [X] Open multiple secondary battle lists
   - [X] UI Double Check
- [X] Skills
   - [X] Reset Experience counter
   - [X] All Skills hide/show flags submenu
   - [X] New window Btn opens Character cyclopedia info
- [X] VIP List
   - [X] Icons, Groups, Ui, Buttons, Style, Floating Screens
   - [X] Right click shows submenu, create groups, new VIPS, edit them, etc.
   - [X] Add new group window UI
   - [ ] Edit VIP new UI --TODO
- [ ] Containers/Inventories
   - [X] SubMenu
   - [X] Sizes, Ui, Buttons, Style
   - [ ] Orderings:
     - [X] Asc/Desc Sorting Features (ALL Except Weight)
     - [ ] Asc/Desc by Weight --TODO
     - [X] Sort Containers First
     - [X] Sort Nested Containers
     - [X] Use Manual Sort Mode
     - [ ] Move contents to 'Obtain' Containers --TODO
     - [ ] Move Nested Containers --TODO
- [X] Bestiary Tracker
- [X] Bostiary Tracker
- [X] Imbuiment Tracker
   - [X] Imbuiment improve to show empty slots placeholder
- [X] Prey Tracker
- [X] Unjustified Points
   - [X] UI, organization, new filling bar.
- [X] QuestLog Tracker 
   - [X] Added Multiple quest tracked being shown.
   - [X] Added Link for each Quest Tracked item directly to the respective Quest Log description.
   - [X] Functions of submenu - Remove All Quests, Remove Completed Quests, Automatically Track New Quests, Automatically untrack New Quests.
- [X] Bot
   - [ ] vBot Analyzer Windows --TODO

### **EXTRA**
- [X] Inventory fix about hands/pvp situations

### **Expected**

## Fixes
WIP - Video showcase.

\# (issue)

## Type of change

Please delete options that are not relevant.

  - [ ] Bug fix (non-breaking change which fixes an issue)
  - [X] New feature (non-breaking change which adds functionality)
  - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - [ ] This change requires a documentation update

## How Has This Been Tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

  - [X] Battle List Multi-Instances Attack behaviour
  - [X] Battle List Saving on Exit/Logout
  - [X] Destroy Battle List Instances
  - [X] Principal Battle List always ID 0
  - [X] Rename each instance
  - [X] Different Filters for each instance
  - [X] Ordering is different for each instance
  - [X] Save, Minimize, Lock, Filter, Open/Closed
  - [X] New Icon for Instances
  - [X] Cap 11 Chars to the window name
  - [X] Old Protocol

**Test Configuration**:

  - Server Version: CANARY 1412
  - Client: OTCLIENT
  - Operating System: WIN 11

## Checklist

  - [X] My code follows the style guidelines of this project
  - [X] I have performed a self-review of my own code
  - [X] I checked the PR checks reports
  - [X] I have commented my code, particularly in hard-to-understand areas
  - [X] I have made corresponding changes to the documentation
  - [X] My changes generate no new warnings
  - [X] I have added tests that prove my fix is effective or that my feature works

libergod committed:
Add miniwindow header image and battle style configuration

- Introduced a new miniwindow header image for UI.
- Created a new style file for the miniwindow battle interface, defining layout, buttons, and scrollable content.
- Configured button states for close, minimize, and toggle filter functionalities.
- Established visual elements such as borders, backgrounds, and dimensions for a cohesive design.

kokekanon commented:
feedback:
# 1: .otui
- there is no need to create a new otui(data/styles/30-miniwindow-battle.otui), 
(in case you need to make a new otui, make sure not to repeat the style name).
<img width="746" height="99" alt="image" src="https://github.com/user-attachments/assets/c82212ac-f64f-4001-9fdb-2238b422c835" />

you should modify data/styles/30-miniwindow.otui
as this would apply to all miniwindows and not only battlelist

example:


<img width="441" height="825" alt="image" src="https://github.com/user-attachments/assets/3b381fc9-a3a1-47f4-8329-2848c265b52d" />


# 2  : horizontalbox to grid
instead of using horizontal and 2 panels, you could make only 1 panel(filterRow1) with grid.
example

```
      layout:
        type: grid
        flow: true
        cell-spacing: 6
        num-columns: 7
```
<img width="528" height="343" alt="image" src="https://github.com/user-attachments/assets/444db294-249d-439c-bf8a-2c263cb97c16" />

result:

<img width="181" height="88" alt="image" src="https://github.com/user-attachments/assets/b096e208-ee79-4d34-b5f1-73fb0dc67695" />


# 3: use constants


<img width="631" height="465" alt="image" src="https://github.com/user-attachments/assets/a8f48c15-03e7-4735-aba1-d9290a7d1269" />

in modules\gamelib\creature.lua: 

```lua
VocationsClient = {
    None = 0,
    Knight = 1,
    Paladin = 2,
    Sorcerer = 3,
    Druid = 4,
    Monk = 5,

    EliteKnight = 11,
    RoyalPaladin = 12,
    MasterSorcerer = 13,
    ElderDruid = 14,
    ExhaltedMonk = 15,
    
}
function Creature.isDruid(self)
    local vocation = self:getVocation()
    return vocation == VocationsClient.Druid or vocation == VocationsClient.ElderDruid
end

function Creature.isSorcerer(self)
    local vocation = self:getVocation()
    return vocation == VocationsClient.Sorcerer or vocation == VocationsClient.MasterSorcerer
end

function Creature.isPaladin(self)
    local vocation = self:getVocation()
    return vocation == VocationsClient.Paladin or vocation == VocationsClient.RoyalPaladin
end

function Creature.isKnight(self)
    local vocation = self:getVocation()
    return vocation == VocationsClient.Knight or vocation == VocationsClient.EliteKnight
end

function Creature.isMonk(self)
    local vocation = self:getVocation()
    return vocation == VocationsClient.Monk  or vocation == VocationsClient.ExhaltedMonk
end
```

`creature:isPlayer() and creature:isKnight() `
apply filter

# 4 old protocol? 

what would happen to the old protocols that do not send vocation? is the button hidden?

libergod commented:
> feedback:
> 
> # 1: .otui
> * there is no need to create a new otui(data/styles/30-miniwindow-battle.otui),
>   (in case you need to make a new otui, make sure not to repeat the style name).
> 
> <img alt="image" width="746" height="99" src="https://private-user-images.githubusercontent.com/114332266/473719133-c82212ac-f64f-4001-9fdb-2238b422c835.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQ0MTY1MjksIm5iZiI6MTc1NDQxNjIyOSwicGF0aCI6Ii8xMTQzMzIyNjYvNDczNzE5MTMzLWM4MjIxMmFjLWY2NGYtNDAwMS05ZmRiLTIyMzhiNDIyYzgzNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgwNVQxNzUwMjlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04MjEwMmVhMDU4NTBiMjhlZDY5OTQ3NDI1NzE4NzQ2NTdkOTM1NWU2Y2M1NmVhYTcwMTBjMTBmMWJkZDUzOGRhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.b7iczsBjgCEZ2uIQNBj3agAf3vI0i1YIdogmTg_Tp3E">
> you should modify data/styles/30-miniwindow.otui as this would apply to all miniwindows and not only battlelist
> 
> example:
> 
> <img alt="image" width="441" height="825" src="https://private-user-images.githubusercontent.com/114332266/473719053-3b381fc9-a3a1-47f4-8329-2848c265b52d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQ0MTY1MjksIm5iZiI6MTc1NDQxNjIyOSwicGF0aCI6Ii8xMTQzMzIyNjYvNDczNzE5MDUzLTNiMzgxZmM5LWEzYTEtNDdmNC04MzI5LTI4NDhjMjY1YjUyZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgwNVQxNzUwMjlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04ODRjMzk0ZmExNDZhNGExZTMzZTRiOTdhMjI0MzFhNzU5ODVmNWExMWM4NmE0YTEwMjQzNTdiN2Y1NGQ3OTQ2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.t4L1v-dpVgbNWTaY-jx9D6Ho_9SCBA_c1Ffm_i43Orw">
> # 2 : horizontalbox to grid
> instead of using horizontal and 2 panels, you could make only 1 panel(filterRow1) with grid. example
> 
> ```
>       layout:
>         type: grid
>         flow: true
>         cell-spacing: 6
>         num-columns: 7
> ```
> 
> <img alt="image" width="528" height="343" src="https://private-user-images.githubusercontent.com/114332266/473719294-444db294-249d-439c-bf8a-2c263cb97c16.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQ0MTY1MjksIm5iZiI6MTc1NDQxNjIyOSwicGF0aCI6Ii8xMTQzMzIyNjYvNDczNzE5Mjk0LTQ0NGRiMjk0LTI0OWQtNDM5Yy1iZjhhLTJjMjYzY2I5N2MxNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgwNVQxNzUwMjlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kMDk5NjQxNTNiNjg4MDUyN2UzNmJiYzQ0ZTJjNmFjNGFmODE0OTBhNjFkN2VhM2VlNjgxNWM4MWIyYjlmNWFiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.0Go9ScY29j0CoteyXYTXSAqwy5wl83Vl3kTRDWXKUVs">
> result:
> 
> <img alt="image" width="181" height="88" src="https://private-user-images.githubusercontent.com/114332266/473719310-b096e208-ee79-4d34-b5f1-73fb0dc67695.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQ0MTY1MjksIm5iZiI6MTc1NDQxNjIyOSwicGF0aCI6Ii8xMTQzMzIyNjYvNDczNzE5MzEwLWIwOTZlMjA4LWVlNzktNGQzNC1iNWYxLTczZmIwZGM2NzY5NS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgwNVQxNzUwMjlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kOGZhMjg1MThkYTI0ZmJjMDVjMjA2ZDcyMmEwMmIwNzkyOGRkNzc2NjlhOGVlZDcyYTU2MmE5NmRhYmMyNzBlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.CRnNo7N2s9TO-F4QyUw5pidf1sEmgopZoeQH82TFAdE">
> # 3: use constants
> <img alt="image" width="631" height="465" src="https://private-user-images.githubusercontent.com/114332266/473719490-a8f48c15-03e7-4735-aba1-d9290a7d1269.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQ0MTY1MjksIm5iZiI6MTc1NDQxNjIyOSwicGF0aCI6Ii8xMTQzMzIyNjYvNDczNzE5NDkwLWE4ZjQ4YzE1LTAzZTctNDczNS1hYmExLWQ5MjkwYTdkMTI2OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgwNVQxNzUwMjlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xMjkxYTkyZDE4YzQ1YjMxMTgyOTZjZTdmZjljYTM4NmNkYzZhODIyZGFlN2FhYmZkZWFjYzc2YjdmZmYyYzlmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.DtK7_NpSh41I6jlt3bXJbU--hYxkMYFIe1lyziYpfPc">
> in modules\gamelib\creature.lua:
> 
> ```lua
> VocationsClient = {
>     None = 0,
>     Knight = 1,
>     Paladin = 2,
>     Sorcerer = 3,
>     Druid = 4,
>     Monk = 5,
> 
>     EliteKnight = 11,
>     RoyalPaladin = 12,
>     MasterSorcerer = 13,
>     ElderDruid = 14,
>     ExhaltedMonk = 15,
>     
> }
> function Creature.isDruid(self)
>     local vocation = self:getVocation()
>     return vocation == VocationsClient.Druid or vocation == VocationsClient.ElderDruid
> end
> 
> function Creature.isSorcerer(self)
>     local vocation = self:getVocation()
>     return vocation == VocationsClient.Sorcerer or vocation == VocationsClient.MasterSorcerer
> end
> 
> function Creature.isPaladin(self)
>     local vocation = self:getVocation()
>     return vocation == VocationsClient.Paladin or vocation == VocationsClient.RoyalPaladin
> end
> 
> function Creature.isKnight(self)
>     local vocation = self:getVocation()
>     return vocation == VocationsClient.Knight or vocation == VocationsClient.EliteKnight
> end
> 
> function Creature.isMonk(self)
>     local vocation = self:getVocation()
>     return vocation == VocationsClient.Monk  or vocation == VocationsClient.ExhaltedMonk
> end
> ```
> 
> `creature:isPlayer() and creature:isKnight() ` apply filter
> 
> # 4 old protocol?
> what would happen to the old protocols that do not send vocation? is the button hidden?

Hello Mate! Thanks for the comments, will take them into consideration for the final version. Since I don't know much about the "rules" used in this project, it was really helpful.

libergod committed:
fix: Window subMenu Setup

libergod committed:
feat: Enhance battle filter system with sorting options and UI adjustments

libergod committed:
feat: Implement battle list name editing functionality and UI components

libergod committed:
Improve: Use constants (feedback 3)

libergod committed:
refactor: Update miniwindow button styles and image handling

libergod committed:
Add corelib as a dependency for the battle module, feat new instance windows

libergod committed:
feat: Enhance BattleListManager with auto-save functionality and lock state management

libergod committed:
feat: Enhance BattleListManager to save and restore lock state for instances

libergod committed:
feat: Add new window button for instance creation and improve safety checks in battle button updates

libergod committed:
Refactor battle.lua: Improve code readability and performance

- Consolidated utility functions for table operations.
- Streamlined filter settings and battle list management.
- Enhanced instance state restoration with minimized window handling.
- Optimized event handling for creature updates across multiple instances.
- Removed redundant comments and legacy code for clarity.
- Improved sorting and filtering logic for battle buttons.

libergod committed:
feat: Update battle list instance window icon and title truncation

libergod committed:
feat: Enhance miniwindow UI with new buttons and improved layout, including toggleFilter and contextMenu buttons

libergod committed:
feat: Add onClick handler for newWindowButton to open Cyclopedia Character tab and enhance skills menu with new options

libergod commented:
Update: Skills tab configured and fixed to hide/show bars and stats.
<img width="286" height="644" alt="image" src="https://github.com/user-attachments/assets/05d91cc8-3762-4d78-af62-46d2554d5cfc" />
<img width="379" height="857" alt="image" src="https://github.com/user-attachments/assets/deb0cb1b-920f-4837-a25d-b723989614e0" />
<img width="299" height="627" alt="image" src="https://github.com/user-attachments/assets/569bc5e2-20f0-48c6-b026-06277a91aae1" />
<img width="621" height="721" alt="image" src="https://github.com/user-attachments/assets/1cc3a66f-869e-4811-9e56-83f09965ea33" />

libergod committed:
feat: Implement context menu for skills with visibility toggles and reset experience option

libergod committed:
feat: Enhance VIP list UI with new FloatingInputWindow and improved layout for add VIP and add group functionalities

libergod committed:
Refactor translations and improve experience tooltip handling

- Updated translation strings for hiding offline VIPs in neededtranslations.lua.
- Enhanced experience speed calculation and tooltip display in skills.lua, ensuring accurate updates based on player state and experience data.
- Modified addvip.otui to improve user prompt for entering character names.
- Adjusted VIP list context menu options to consistently refer to offline VIPs.

libergod committed:
Refactor container UI and enhance pagination functionality

- Updated the container UI layout in `container.otui` to improve visual hierarchy and usability.
- Changed `PageButton` to inherit from `UIButton` for better styling and interaction.
- Introduced new UI elements such as `miniwindowIcon` and `miniwindowTitle` for enhanced window management.
- Implemented a `separator` and `contentsPanel` to better organize container items and navigation.
- Enhanced the `toggleContainerPages` function to dynamically adjust UI elements based on pagination state.
- Improved button visibility and positioning logic for `prevPageButton` and `nextPageButton` based on current page.
- Added event handlers for minimizing and maximizing the container window to manage page visibility.
- Updated item navigation logic to preserve container height during page transitions.
- Added new images for pagination buttons to support the updated UI design.

libergod committed:
feat: Enhance container sorting options and context menu functionality

libergod committed:
feat: Add totalSlots to ImbuementTrackerItem and update related UI components

libergod committed:
Enhance Bestiary and Bosstiary Tracker Functionality

- Implemented character change detection to clear and reload tracker data.
- Improved initialization and caching of tracker data for better performance.
- Added context menu for sorting options in both Bestiary and Bosstiary trackers.
- Enhanced UI elements including title length limits and icon settings for trackers.
- Introduced functions for saving and loading tracker filters and data specific to characters.
- Added fallback mechanisms for refreshing tracker data when windows are opened.
- Updated progress bar visuals for better user experience.
- Added debug functions to monitor tracker state and data.
- Included new icons for Bestiary and Bosstiary trackers.

libergod committed:
feat: Update localization files and enhance unjustified points UI with new assets

libergod committed:
Add Prey Tracker functionality and localization updates

- Implemented Prey Tracker feature in the game, including UI adjustments and new functionality for tracking prey.
- Added localization for Polish, Portuguese, and Swedish languages, including translations for "Prey", "Prey Tracker", and related terms.
- Updated the UI to hide unnecessary buttons and set up a new miniwindow title and icon for the Prey Tracker.
- Introduced new image asset for the Prey Tracker icon.

libergod committed:
feat: Update Quest Tracker UI elements and improve layout

libergod committed:
feat: Enhance quest tracking functionality and UI interactions

libergod committed:
feat: Implement auto-untrack for completed quests and enhance quest tracker UI

libergod committed:
feat: Refactor battle UI components and improve scrollbar behavior in battle instances

libergod committed:
feat: Update visibility logic for BattleMonks and adjust margin for MiniWindow based on client version

libergod committed:
feat: Improve battle UI logging and refactor filter panel layout for better organization

Fix issues from sonar, change behavior of battle list to verticalbox layout. Clean organizational functions not needed anymore. Cleanup any debug infos/logs not needed anymore.

libergod committed:
feat: Rename ChangeListName to MainWindow for clarity in UI structure

libergod committed:
feat: Update expert button images and adjust unjustified points progress bar logic for improved UI and functionality

libergod committed:
fix: Bug that prevented to remove target (UI update).

libergod committed:
Update unjustifiedpoints.lua

libergod committed:
Update unjustifiedpoints.lua

kokekanon commented on init.lua:
```diff
@@ -104,7 +104,7 @@ local function loadModules()
 
     -- uncomment the line below so that modules are reloaded when modified. (Note: Use only mod dev)
-    -- g_modules.enableAutoReload()
+    g_modules.enableAutoReload()
```
delete this

kokekanon commented on modules/game_battle/battle.otmod:
```diff
@@ -4,6 +4,7 @@ Module
   author: andrefaramir, BeniS
   website: https://github.com/edubart/otclient
   sandboxed: true
+  dependencies: [ corelib ]
```
this does not seem necessary

kokekanon commented on modules/game_unjustifiedpoints/unjustifiedpoints.lua:
```diff
@@ -31,17 +36,68 @@ function init()
+        lockButton:setMarginTop(0)
+    end
+
+    unjustifiedPointsButton = modules.game_mainpanel.addToggleButton('unjustifiedPointsButton',
```
If you change this from gameStart to onInit, it will cause the button to appear for all versions — for example, it would show up in 8.6, even though that version doesn't have that feature.

<img width="176" height="81" alt="image" src="https://github.com/user-attachments/assets/41b29822-99aa-4cf9-adad-d12d40619f10" />

libergod replied:
Sorry, misunderstood your point. Should we move something to another function ? 

unjustifiedPointsButton = modules.game_mainpanel.addToggleButton('unjustifiedPointsButton',
        tr('Unjustified Points'), '/images/options/button_frags', toggle)
    unjustifiedPointsButton:setOn(true)

kokekanon commented on modules/game_unjustifiedpoints/unjustifiedpoints.lua:
```diff
@@ -96,11 +153,8 @@ function toggle()
 
 function online()
-    if g_game.getFeature(GameUnjustifiedPoints) and not unjustifiedPointsButton then
+    if g_game.getFeature(GameUnjustifiedPoints) then
```
If you change this from gameStart to onInit, it will cause the button to appear for all versions — for example, it would show up in 8.6, even though that version doesn't have that feature.

<img width="176" height="81" alt="image" src="https://github.com/user-attachments/assets/7590dd7d-507d-4189-96bc-afa7137d1864" />

libergod replied:
Sorry, misunderstood your point. Should we move something to another function ?

kokekanon replied:
Previously, it was like this:
in onGameStart + a function with `if g_game.getFeature(GameUnjustifiedPoints)`

If the version you connect to the server with has the GameUnjustifiedPoints feature, then it creates the button.
If it doesn’t have the feature, it simply doesn’t create the button.

**In your code, you changed the button creation to onInit(),**
which creates the button for all versions.


 This behavior is not correct, because for example, in version 8.6, the UnjustifiedPoints feature doesn’t exist, but the button is created.

libergod replied:
GOT IT! Thanks for that, fixed.

<img width="1327" height="300" alt="image" src="https://github.com/user-attachments/assets/5b5d7a6b-fd5c-48f9-adea-fd53b508ecd6" />

kokekanon replied:
add and not unjustifiedPointsButton then

libergod replied:
Done.

kokekanon commented on modules/game_battle/battle.lua:
```diff
@@ -66,71 +1267,130 @@ local function disconnecting(gameEvent)
+                if lastCreatureSelected and obj.creature == lastCreatureSelected then
+                    lastCreatureSelected = nil
+                end
+                
```
add -> obj:resetState()

<img width="177" height="324" alt="image" src="https://github.com/user-attachments/assets/431e3488-826e-4fc3-a401-055d8e0332ce" />

libergod replied:
What did you do to cause that behaviour? Had tried exhaustively.

kokekanon replied:
https://github.com/mehah/otclient/pull/1260

look gif https://github.com/mehah/otclient/issues/1245#issuecomment-3099925716

This is solved by adding this.

`obj:resetState()`

libergod replied:
<img width="1166" height="520" alt="image" src="https://github.com/user-attachments/assets/442bd6f5-047f-4d00-953e-63ba4841bc30" />

Fixed. Thanks!

kokekanon commented:
<img width="297" height="202" alt="image" src="https://github.com/user-attachments/assets/f80d6dce-7e66-4396-9692-33bc5a13389b" />
It doesn't have a default behavior.

kokekanon commented on modules/game_skills/skills.lua:
```diff
@@ -260,6 +662,32 @@ end
+    end
+    
+    -- Add onClick handler to newWindowButton to open Cyclopedia Character tab
+    local newWindowButton = skillsWindow:recursiveGetChildById('newWindowButton')
```
if g_game.getClientVersion() < 1310 then

newWindowButton:hide()

libergod replied:
<img width="1281" height="414" alt="image" src="https://github.com/user-attachments/assets/6b497639-0a3a-4333-bc0f-e79e9e930747" />

kokekanon commented on modules/game_skills/skills.lua:
```diff
@@ -100,6 +130,375 @@ function terminate()
+function showSkillsContextMenu(widget, mousePos, mouseButton)
+    local menu = g_ui.createWidget('SkillsListSubMenu')
+    menu:setGameMenu(true)
+    
```
if version < 14.12 then
hide this children
menu.showOffenceStats
menu.showDefenceStats
menu.showMiscStats


if not g_game.getFeature(GameOfflineTrainingTime) then
menu.showOffilineTraining hide()

kokekanon commented on modules/game_viplist/viplist.lua:
```diff
@@ -752,30 +882,106 @@ function createAddGroupWindow()
+
+function destroyAddGroupWindow()
     if addGroupWindow then
+        addGroupWindow:setVisible(false)
```
If the object is going to be destroyed, it doesn't make sense to call setVisible(false).

libergod commented:
> <img alt="image" width="297" height="202" src="https://private-user-images.githubusercontent.com/114332266/482169683-f80d6dce-7e66-4396-9692-33bc5a13389b.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTYyMjM5MTIsIm5iZiI6MTc1NjIyMzYxMiwicGF0aCI6Ii8xMTQzMzIyNjYvNDgyMTY5NjgzLWY4MGQ2ZGNlLTdlNjYtNDM5Ni05NjkyLTMzYmM1YTEzMzg5Yi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgyNlQxNTUzMzJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03M2I2MDMzODNlZmJkMmQ0YjdmNTkyMjkzZmQxOGVjM2MyOTlmMjZiODBhZjYyZjQzNjdmYmZjOGY5YmQwMDZmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.swNLBPRizMhKW7lUW834QSZSy6_Om8kq8rT2w_uHun0"> It doesn't have a default behavior.

Fixed, added default behaviour to Asc By Display Time.

<img width="1009" height="457" alt="image" src="https://github.com/user-attachments/assets/3a63d4bf-4a81-42f5-9eee-6de365d945d5" />

libergod committed:
fix: Ensure unjustified points button state is correctly set on initialization and online status
feat: Hide stats options for older client versions in skills menu and adjust visibility settings

libergod commented:
All recommendations fixed/changed as expected.

libergod committed:
Update init.lua

kokekanon commented on modules/game_cyclopedia/game_cyclopedia.lua:
```diff
@@ -204,15 +351,72 @@ function controllerCyclopedia:onGameStart()
         controllerCyclopedia.ui.CharmsBase.Icon:setImageSource("/game_cyclopedia/images/monster-icon-bonuspoints")
     end
+    
+    -- Initialize cached tracker data for immediate loading with delay to ensure character name is available
```
Move this scheduleEvent block to line 349, inside the true block of (`if g_game.getClientVersion() >= 1310 then`).
<img width="595" height="72" alt="image" src="https://github.com/user-attachments/assets/2b0b7973-cfbf-4def-9960-9cf384071013" />

the game_cyclopedia module does not exist in 8.6

kokekanon commented on modules/game_containers/containers.lua:
```diff
@@ -165,21 +1074,32 @@ function onContainerOpen(container, previousContainer)
+        containerWindow:setContentHeight(cellSize.height * layout:getNumLines())
+    else
+        local filledLines = math.max(math.ceil(container:getItemsCount() / layout:getNumColumns()), 1)
+        containerWindow:setContentHeight(filledLines * cellSize.height)
```
That logic needs to be improved.
`setContentHeight`
<img width="185" height="53" alt="image" src="https://github.com/user-attachments/assets/c60fc03c-af84-4454-a844-441268a1da4b" />

<img width="185" height="53" alt="image" src="https://github.com/user-attachments/assets/7ae17308-ba1f-48e0-ba50-0a35d87e1362" />

libergod replied:
Fixed by adding a minimum height for both different containers (with/without) page counting.

kokekanon commented:
https://github.com/user-attachments/assets/85747337-0fe6-4bc3-9932-9589116047ca

bug:
- I open the miniwindows battlelist and open a battlelist instance.
- I close the original miniwindows and keep the instance open.
- The `local function disconnecting(gameEvent)` is executed, disconnecting the events.


https://github.com/mehah/otclient/blob/bad4e242cfd6bdca187380e2b4405666576fcd65/modules/game_battle/battle.lua#L1278-L1291

- This causes the only miniwindows  open battlelist (the instance) to stop receiving updates.

libergod committed:
fix: Implement container window resize restrictions and improve content height calculation

libergod committed:
fix: Implement window resize restrictions for battle window and filter panel

libergod committed:
fix: Manage event connections for battle list instances to ensure proper handling on open/close

libergod committed:
fix: Set minimum height for imbuement tracker and skills windows to improve UI consistency

libergod commented:
Fixed previous comments:
- [X] Manage Resize of containers, battle list, skills, and imbuement tracker
- [X] Manage instances and battle list updates.
- [X] Reorg code cyclopedia.

https://github.com/user-attachments/assets/aede35ba-f823-456f-b400-2c9ce0fd3001


https://github.com/user-attachments/assets/179566d8-3b64-4405-9d5e-16a4cee41be5


https://github.com/user-attachments/assets/4dd13f6a-2872-401f-af69-be440ff1b8df

kokekanon commented:
https://github.com/user-attachments/assets/4a33b02e-cf7c-44be-9d07-2a87295a6b76

I don't know how Cipsoft behaves, but there should be a destroy or hide , or update the checkbox at click?

libergod commented:
> bandicam.2025-08-26.21-03-20-800.mp4 
> I don't know how Cipsoft behaves, but there should be a destroy or hide, or update the checkbox at click?

Yes it should... Will double check that, cuz I had already implemented a fix for that in different windows. Might have forgotten that one.

EDIT: FIxed.

libergod committed:
fix: Update bestiary tracker sorting filters and default settings; adjust prey tracker dimensions for improved UI

mehah commented:
@libergod 
is done?

libergod commented:
> @libergod is done?

YES sir, waiting final review.
